### PR TITLE
Attachment box style fixes

### PR DESCRIPTION
**Pull request by xdu-odoo**
- Fix filename display on hover
- style "add attachment" button
- style list attachments display

### CODE REVIEW

**Comment by xdu-odoo** [comment]
Red only because of migrations

**Comment by xdu-odoo** [comment]
Red only because of migrations (again)

**Comment by xdu-odoo** [comment]
Green :+1:

**Comment by xdu-odoo** [comment]
green :+1:

File: addons/mail/static/src/owl/components/attachment/attachment.js [hunk 1]
```diff
@@ -141,6 +142,9 @@ Attachment.props = {
     detailsMode: {
         type: String,
     }, //['auto', 'card', 'hover', 'none']
+    imageSize: {
+        type: String,
+    }, //['small', 'medium', 'large']
```
Comment (alexkuhn) [inline addons/mail/static/src/owl/components/attachment/attachment.js]: `imageSize: String, // ['small', 'medium', 'large']`

File: addons/mail/static/src/owl/components/attachment/attachment.xml [hunk 2]
```diff
@@ -16,15 +16,18 @@
             <div class="o_Attachment_image o_image" t-on-click="_onClickImage"
                 t-att-class="{
                     'o-attachment-viewable': env.store.getters.isAttachmentViewable(props.attachmentLocalId),
-                    'o-details-overlay': detailsMode !== 'card'
+                    'o-details-overlay': detailsMode !== 'card',
+                    'o-small': props.imageSize === 'small',
+                    'o-medium': props.imageSize === 'medium',
+                    'o-large': props.imageSize === 'large',
```
Comment (alexkuhn) [inline addons/mail/static/src/owl/components/attachment/attachment.xml]: alphabetical order of classnames

File: addons/mail/static/src/owl/components/attachment_box/attachment_box.scss [hunk 3]
```diff
@@ -6,20 +6,43 @@
...
 }
 
 .o_AttachmentBox_title {
-    text-align: center;
+    display: flex;
+    align-items: center;
+}
+
+.o_AttachmentBox_titleText {
+    padding: 10px;
```
Comment (alexkuhn) [inline addons/mail/static/src/owl/components/attachment_box/attachment_box.scss]: no spacers?

Comment (xdu-odoo) [reply addons/mail/static/src/owl/components/attachment_box/attachment_box.scss]: didn't know about them at this time

**Comment by xdu-odoo** [comment]
Green :+1:

File: addons/mail/static/src/owl/components/attachment_box/attachment_box.scss [hunk 4]
```diff
@@ -6,20 +6,43 @@
...
 
 // ------------------------------------------------------------------
 // Style
 // ------------------------------------------------------------------
 
-.o_AttachmentBox {
-    border-top: 1px dashed #dee2e6;
+.o_AttachmentBox_dashedLine {
+    border-style: dashed;
+    border-color: #dee2e6;
```
Comment (alexkuhn) [inline addons/mail/static/src/owl/components/attachment_box/attachment_box.scss]: no hard-coded colour, use somewhat equivalent variant of `gray()` instead

**Comment by xdu-odoo** [comment]
Red only because of migrations

File: addons/mail/static/src/owl/store/messaging_actions.js [hunk 5]
```diff
@@ -3326,7 +3327,7 @@ const actions = {
         const threadAttachmentLocalIds = thread.attachmentLocalIds;
         if (!threadAttachmentLocalIds || !threadAttachmentLocalIds.includes(attachmentLocalId)) {
             dispatch('_updateThread', threadLocalId, {
-                attachmentLocalIds: threadAttachmentLocalIds.concat([attachmentLocalId]),
+                attachmentLocalIds: [attachmentLocalId].concat(threadAttachmentLocalIds),
```
Comment (alexkuhn) [inline addons/mail/static/src/owl/store/messaging_actions.js:3330]: ok, I may break ordering of attachments in my entity branch, so will need to be keep an eye on that ;)

**Review by alexkuhn — APPROVED** [review]

**Pull request merged**